src/apps/chifra/internal/receipts/output.go

```diff
@@ -46,42 +46,53 @@ func (opts *ReceiptsOptions) ReceiptsInternal() (err error, handled bool) {
 
 	// EXISTING_CODE
 	// TODO: BOGUS THIS WORK IS INCOMPLETE -- ONLY WORKS IF THE ENV VAR IS EXPORTED
-	newnew := os.Getenv("NEW") == "true"
-	if !opts.Articulate && newnew {
-		defer opts.Globals.RenderFooter()
-		err = opts.Globals.RenderHeader(types.SimpleReceipt{}, &opts.Globals.Writer, opts.Globals.Format, opts.Globals.ApiMode, opts.Globals.NoHeader, true)
+	// newnew := os.Getenv("NEW") == "true"
+	// if !opts.Articulate && newnew {
+	// defer opts.Globals.RenderFooter()
+	// err = opts.Globals.RenderHeader(types.SimpleReceipt{}, &opts.Globals.Writer, opts.Globals.Format, opts.Globals.ApiMode, opts.Globals.NoHeader, true)
+	// h := types.SimpleReceipt{}
+
+	// TODO: support options
+	// err = opts.Globals.RenderHeader(h.Model(opts.Globals.Verbose).Data, &opts.Globals.Writer, opts.Globals.Format, opts.Globals.ApiMode, opts.Globals.NoHeader, true)
+	if err != nil {
+		return err, true
+	}
+
+	cnt := 0
+	for idIndex, rng := range opts.TransactionIds {
+		txList, err := rng.ResolveTxs(opts.Globals.Chain)
```

Comment:
Another TODO list item (in other words, don't fix it now, but note it). I've been thinking that what we should do here instead of `Resolving` (which builds an array of transactions ids to traverse), we should stream through this list and do the traversal on the fly. An example of where this breaks down is the block range (which doesn't apply here, but does for `chifra blocks`, for example) of `chifra blocks 0-latest:2` which generates an array with 8,000,000 records. This should be "streamed" as well.

Reply:
Great point and this is something I've been working on, I just don't have the code yet.

Reply:
I feel like this might be a bit more complicated because it touches code that is used by a lot of code. Let's think of this as a "nice to have" and not open this particular can of worms too wide just yet. Unless it's not as complicated as I think it is.